engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -333,6 +313,29 @@ private void addSkyNodes(RenderGraph renderGraph) {
         renderGraph.connect(lastUpdatedGBufferClearingNode, backdropNode, hazeIntermediateNode, hazeFinalNode);
     }
 
+    private void addWorldRenderingNodes(RenderGraph renderGraph) {
+        // Ideally, world rendering nodes only depend on gBufferClearingNode. However, since haze is produced
+        // by blurring the gBuffer and we don't want world objects/lighting to be a part of the haze,
+        // the world rendering nodes need to run after hazeIntermediateNode.
+        Node hazeIntermediateNode = renderGraph.findNode("engine:hazeIntermediateNode");
+
+        Node opaqueObjectsNode = new OpaqueObjectsNode(context);
+        renderGraph.addNode(opaqueObjectsNode, "opaqueObjectsNode");
+        renderGraph.connect(hazeIntermediateNode, opaqueObjectsNode);
+
+        Node opaqueBlocksNode = new OpaqueBlocksNode(context);
+        renderGraph.addNode(opaqueBlocksNode, "opaqueBlocksNode");
+        renderGraph.connect(hazeIntermediateNode, opaqueBlocksNode);
+
+        Node alphaRejectBlocksNode = new AlphaRejectBlocksNode(context);
+        renderGraph.addNode(alphaRejectBlocksNode, "alphaRejectBlocksNode");
+        renderGraph.connect(hazeIntermediateNode, alphaRejectBlocksNode);
+
+        Node overlaysNode = new OverlaysNode(context);
+        renderGraph.addNode(overlaysNode, "overlaysNode");
+        renderGraph.connect(hazeIntermediateNode, overlaysNode);
+    }
+
```

Comment:
One minor note about the commit list: you have made all the changes in a single commit. 

Ideally, simple changes in the order of methods or in the order of lines of codes are done in their own, trivial commits. More functional commits, i.e. bugfixes or code additions, are then done in other commits. 

This is to avoid the situation we have above, where the simple change in position of the addWorldRenderingNodes method is overlapping with the changed connections to the world rendering nodes.

Reply:
Noted, I'll keep that in mind next time.